### ISSUE TITLE
install.jl fails in a shared installation (rmdir: Permission denied)

### ISSUE THREAD

**Issue by AndersBlomdell** [jheinen/GR.jl#348]
I'm trying to install julia in a directory shared by multiple users (write protected), but GR/<slug>/deps/build.jl fails, due to the fact that build.jl does not test for existing versions of grdir in  GR/<slug>/deps/gr, changing the line 

        for d in (homedir(), "/opt", "/usr/local", "/usr")

to:

        for d in (homedir(), pwd(), "/opt", "/usr/local", "/usr")

but I don't know the pwd() is in the right place in the list, or if there is a better solution to my problem.

**Comment by jheinen** [comment]
You may use `ENV["GRDIR"]=pwd()` before installing GR.

**Comment by AndersBlomdell** [comment]
Yes, but it will not be the same location, since `pwd()` in build.jl will be the one given by the sandbox it's running in, i.e. matching `grdir = joinpath(dirname(@__FILE__), "..", "deps", "gr")` inn `GR.jl`.

But the right solution is probably to make use of Artifacts.jl, since I expected the files to be in an `artifacts` directory
